astropy/cosmology/units.py

```diff
@@ -62,13 +64,27 @@ def redshift_temperature(cosmology=None, **atzkw):
         A cosmology realization or built-in cosmology's name (e.g. 'Planck18').
         If None, will use the default cosmology
         (controlled by :class:`~astropy.cosmology.default_cosmology`).
+    distance_kind : {'comoving', 'lookback', 'luminosity'} or None, optional
+        The distance type for the Equivalency.
+        Note this does NOT include the angular diameter distance as this
+        distance measure is not monotonic.
     **atzkw
         keyword arguments for :func:`~astropy.cosmology.z_at_value`
 
     Returns
     -------
     `~astropy.units.equivalencies.Equivalency`
         Equivalency between redshift and temperature.
+
+    Examples
+    --------
+    >>> import astropy.units as u
+    >>> import astropy.cosmology.units as cu
+    >>> from astropy.cosmology import WMAP9
+
+    >>> z = 1100 * cu.redshift
+    >>> z.to(u.Mpc, cu.redshift_distance(WMAP9, distance="comoving"))  # doctest: +FLOAT_CMP
```

Comment:
How can this test work, since the argument name is not `distance_kind`? I think this will go in `**atkw` and use the default instead...

Reply:
Yikes. Good thing the default is "comoving"!